hack/lib.sh

```diff
@@ -531,14 +531,29 @@ provider_disabled() {
   # e.g. "VSPHERE_E2E_DISABLED"
   local disableEnv="${1^^}_E2E_DISABLED"
   local labelName="test/require-$1"
+  local branchName="${2:-}"
 
-  # tests can be globally disabled by having a special environment
+  # Tests can be globally disabled by having a special environment
   # variable injected via the Prow preset; if they are not disabled,
   # we are done here.
+
   if [ -z "${!disableEnv:-}" ]; then
     return 1
   fi
 
+  # To prevent breaking existing releases, this check does apply to
+  # release branches. On those every failed test needs to be manually
+  # overridden.
+
+  if [ -z "$branchName" ]; then
+    branchName="${PULL_BASE_REF:-main}"
+  fi
+
+  if [[ "$branchName" =~ release/.* ]]; then
+    echodate "\$$disableEnv is set, but in a release branch ($branchName) this has no effect."
+    return 1
+  fi
```

Comment:
won't this be a problem since we disable tests via prow presets, which should be used by release branch CI as well?

Reply:
I don't understand.

Reply:
I suspect I simply don't understand the context of this code, but from my perspective it looks like "check if you are on a release branch and fail if environments are disabled". But since environments are disabled in a prow preset, this would be inherited into release branches running CI jobs and referencing that preset (since we don't have dedicated release presets), thus CI on release branches would fail.

As said, maybe I just don't understand what's happening, but hopefully this explained my concern a bit better?

Reply:
When the preset is configured to disable a certain provider, we do not want this to have any effect on release branches. This means we want `provider_disabled` to return `false`, i.e. anything non-zero. And so the function checks if the PR's target branch is a release branch. If so, it returns 1 (false) and so the `if provider_disabled; then exit 0; fi` check is not triggered and so the job continues.

Reply:
ahhh. okay. Now I get it, thanks for explaning.